molt/src/expr.rs

```diff
@@ -33,57 +35,42 @@ pub(crate) enum Type {
 /// I could have used a union to save space, but we don't keep large numbers of these
 /// around.
 #[derive(Debug, PartialEq)]
-pub(crate) struct Datum {
-    vtype: Type,
-    int: MoltInt,
-    flt: MoltFloat,
-    str: String,
+pub(crate) enum Datum {
+    Int(MoltInt),
+    Float(MoltFloat),
+    String(String),
 }
 
 impl Datum {
     fn none() -> Self {
-        Self {
-            vtype: Type::String,
-            int: 0,
-            flt: 0.0,
-            str: String::new(),
-        }
+        Datum::String("".to_owned())
     }
 
     pub(crate) fn int(int: MoltInt) -> Self {
-        Self {
-            vtype: Type::Int,
-            int,
-            flt: 0.0,
-            str: String::new(),
-        }
+        Datum::Int(int)
     }
 
     pub(crate) fn float(flt: MoltFloat) -> Self {
-        Self {
-            vtype: Type::Float,
-            int: 0,
-            flt,
-            str: String::new(),
-        }
+        Datum::Float(flt)
     }
```

Comment:
These methods don't really contain any useful functionality anymore, it might be a good idea to replace any calls to them with direct invocations of the enum variant constructors